cmd/secrets-provider/main.go

```diff
@@ -87,10 +87,13 @@ func main() {
 	}
 
 	// Provide secrets
-	err = provideSecrets()
-	if err != nil {
-		errStr := fmt.Sprintf(messages.CSPFK039E, secretsConfig.StoreType, err.Error())
-		logError(errStr)
+	go provideSecrets()
+	select {
+	case err = <-secrets.SecretProviderError:
```

Comment:
Looking at this again, I think the logic of calling a goroutine and providing an error channel should be moved to the `pkg/secrets` package. Generally, we want the `main.go` to be as simple as possible, and I think that this implementation detail for periodic secrets provider function should be abstracted away from `main.go`.

We would still need to do exactly what you're doing here inside the test code, since that will need to have a goroutine that can either run once and return, run once and sleep, or run periodically... and be able to stop that goroutine from the test thread.

To see what I'm suggesting, take a look at these diffs that I created on a branch based on your PR branch:
https://github.com/cyberark/secrets-provider-for-k8s/commit/d61c96c091f171be0aa04be5c75a6407a3e5f092

Reply:
Patched in the diffs.